[Doc] add known limitation and settings for connection issue (#167)
logstash-plugins/logstash-integration-jdbc: docs/input-jdbc.asciidoc

}
---------------------------------------------------------------------------------------------------

==== Known limitations

> Theoretically, this plugin can function with any JDBC driver.

Reviewer: ```suggestion
The JDBC input plugin leverages the "sequel" library to work with databases through their JDBC drivers.
```

> However, the implementation of drivers may vary, potentially leading to unexpected behavior in connection pool management.

Reviewer: ```suggestion
However, the implementation of drivers may vary, potentially leading to unexpected behavior.
```

> ===== Unable to reuse connection

Reviewer: ```suggestion
===== Unable to reuse connections
```

> Stale connections may return to the pool, causing the plugin reuse closed connections. Sybase is affected by this issue.

Reviewer: Some databases - such as Sybase or SQL Anywhere - may have issues with stale connections, timing out between scheduled runs and never reconnecting.

> Stale connections may return to the pool, causing the plugin reuse closed connections. Sybase is affected by this issue.

Reviewer: ```suggestion

Some databases - such as Sybase or SQL Anywhere - may have issues with stale connections, timing out between scheduled runs and never reconnecting.
```

> One solution is to enable <<plugins-{type}s-{plugin}-jdbc_validate_connection>> and set <<plugins-{type}s-{plugin}-jdbc_validation_timeout>> to a shorter interval than the <<plugins-{type}s-{plugin}-schedule>>.

Reviewer: ```suggestion
To ensure connections are valid before queries are executed, enable <<plugins-{type}s-{plugin}-jdbc_validate_connection>> and set <<plugins-{type}s-{plugin}-jdbc_validation_timeout>> to a shorter interval than the <<plugins-{type}s-{plugin}-schedule>>.
```

[source,ruby]
---------------------------------------------------------------------------------------------------
input {
  jdbc {
    schedule => "* * * * *"       # run every minute
    jdbc_validate_connection => true

>     jdbc_validation_timeout => 55 # 55 seconds

Reviewer: ```suggestion
    jdbc_validation_timeout => 50 # 50 seconds
```
